compare.go

```diff
@@ -0,0 +1,186 @@
+package linq
+
+type comparer func(interface{}, interface{}) int
+
+// Comparable is an interface that has to be implemented by a
+// custom collection elememts in order to work with linq.
+//
+// Example:
+//		func (f foo) CompareTo(c Comparable) int {
+//			a, b := f.f1, c.(foo).f1
+//
+//			if a < b {
```

Comment:
they seem pretty the same legth since I use gofmt
			switch {
			case a > b:
				return -1
			case b < a:
				return 1
			default:
				return 0
			}
vs
			if a < b {
				return -1
			} else if a > b {
				return 1
			}

			return 0

There are other variants as well. Consider:
			if a < b {
				return -1
			}
			if a > b {
				return 1
			}
			return 0
WDYT?